server/src/main/java/io/druid/segment/realtime/appenderator/Appenderator.java

```diff
@@ -61,8 +61,6 @@
    * This method may trigger a {@link #persistAll(Committer)} using the supplied Committer. If it does this, the
    * Committer is guaranteed to be *created* synchronously with the call to add, but will actually be used
    * asynchronously.
-   * <p>
-   * The add, clear, persist, persistAll, and push methods should all be called from the same thread.
```

Comment:
```add()```, ```persist()```, ```push()``` and ```drop()``` methods can be called by different threads in this patch. ```push()``` and ```persist()``` can be called by AppenderatorDriver.publishExecutor, and ```drop()``` is called by CoordinatorBasedSegmentHandoffNotifier.scheduledExecutor. Actually, ```persist()``` is called by AppenderatorImpl.pushExecutor even before this patch.